### DISCUSSION TITLE
Example with 2D Materials

### DISCUSSION THREAD

**Discussion by ftherrien** [ftherrien/VaspGibbs#2] [VaspGibbs Forum]
Comment from @shahid-sattar from [this issue](https://github.com/ftherrien/VaspGibbs/issues/1#issuecomment-1120487073):

> It would be great if you kindly post an example, let's say for the reaction
> 2D MoS2 (solid) + 2 Se (gas) ----> 2D MoSe2 (solid) + 2 S (gas)
> using vasp_gibbs that can clarify the whole procedure step by step with analysis of energies.
> It would be a great great help.

**Comment by shahid-sattar** [comment]
It’s very good that you moved it to discussion. 
In fact I’m thinking to provide results and input to post here, not just one but more examples to help any potential user.

**Reply by ftherrien** [reply]
That would be awesome!

**Comment by ftherrien** [comment]
I am not familiar with the thermodynamics of 2D structures. So the harmonic oscillator approximation may not be valid and, depending on the environment, you might have to consider rotation and translations in and around the z axis. (If you find the theory on that and would like to implement it in VaspGibbs I would be happy to help!) 

Here is how I would naively do it with the current code:

1. In the MoS2 directory with a completed Vasp calculation run:
`vasp_gibbs`

2. In the Se directory with a completed Vasp calculation run:
`vasp_gibbs --molecule`

3. In the MoSe2 directory with a completed Vasp calculation run:
`vasp_gibbs`

4. In the S directory with a completed Vasp calculation run:
`vasp_gibbs --molecule`

**When I say: "a completed Vasp calculation" I mean an SCF calculation or a relaxation. You do not need to run a frequency calculation yourself, `vasp_gibbs` will set up that calcuation for you and rerun Vasp for you.** However, if you do provide `vasp_gibbs` with a frequency calculation it will recognize it and use your results. 


Then for each of the calculations you can find the gibbs free energy at ambient conditions in the `VaspGibbs.md` file. If you would like the free energy at a different temperature, say 650K just run:
`vasp_gibbs -T 650`
or
`vasp_gibbs --molecule -T 650`

**Comment by shahid-sattar** [comment]
Alright. Having a single S atom in a gas phase (S atom in a box), I first did frequency calculation (NSW = 100,IBRION = 5, NFREE  = 2) and used vasp_gibbs on top, it worked and I can clearly see the terms in results. Though there is some confusion why E_DFT is not added in G but subtracted here but at least I get the result. 
![Screenshot from 2022-05-12 13-42-33](https://user-images.githubusercontent.com/61748682/168067284-36e99507-2f31-49fc-9d63-f7066a0e60b4.png)

Now, the confusion remains with 2D MoS2 (solid) as to how frequency calculation should be done? Is it that I should use unit cell and simply do the same as above  (NSW = 100,IBRION = 5, NFREE  = 2) or take some supercell and do the phonon calculation? 
Because running vasp_gibbs on top of an scf calculation for MoSe2 (solid) does not work and gives the following error message.
![Screenshot from 2022-05-12 13-51-01](https://user-images.githubusercontent.com/61748682/168068736-2569ce11-8661-4708-a83a-8715c3e02444.png)

**Reply by shahid-sattar** [reply]
Your quick response would be extremely useful. Thanks

**Reply by ftherrien** [reply]
For the first part: 
> Though there is some confusion why E_DFT is not added in G but subtracted here but at least I get the result.

E_DFT **is** included in the Gibbs Free energy. I added G - E_DFT to show what is the total difference (or correction) from considering Gibbs Free energy instead of simply the DFT energy. Does that make sense?

As for the second part of your question. Is it possible that IBRION is specified twice in your INCAR? It seems like it set IBRION to 5 but the frequency calculations did not happen. Can you post `INCAR` and `INCAR.save` here?

**Reply by shahid-sattar** [reply]
Thanks for the response.
Yes, now it is clear that E_Gibbs is the new energy incorporating the Temp effect in E_dft. So, we can simply use this value on the left and right side of the equation to evaluate the change in Gibbs free energy(E(products)-E_(reactants)).

**Reply by shahid-sattar** [reply]
One last part, this change (which I think is mainly the vibrational contribution for a solid) is for the whole supercell if I use a 2*2*2 supercell. We have to divide accordingly to get the change for the unit cell. Right?

**Reply by ftherrien** [reply]
Yes that's correct you need to divide by 8 to get the Gibbs Free energy per unit cell
